### DISCUSSION TITLE
Formatting of Get-FalconVulnerability command

### DISCUSSION THREAD

**Discussion by cjlane0289** [CrowdStrike/psfalcon#92] [Q&A] [answered]
When I get the Spotlight Vulnerability reports directly from the Falcon Console, I see all of the fields. When I get them using PSFalcon's Get-FalconVulnerability command, a number of the fields are grouped into JSONs.  The following is an example of 3 fields.

The following is the line of code I am using:
Get-FalconVulnerability -All -Detailed -Filter $Filter | Export-Csv -Path $vulnerabilities -NoTypeInformation

I have attach a sample of the report from PS Falcon, the code I am using, and a sample of the report from the Falcon Console.

How do I get the data from Get-FalconVulnerability to look like the Falcon Console report?
[PSFalcon Sample.xlsx](https://github.com/CrowdStrike/psfalcon/files/6822983/PSFalcon.Sample.xlsx)
[PSFalcon Code.txt](https://github.com/CrowdStrike/psfalcon/files/6823000/PSFalcon.Code.txt)
[FalconConsole Sample.xlsx](https://github.com/CrowdStrike/psfalcon/files/6823019/FalconConsole.Sample.xlsx)

**Comment by bk-cs** [comment, marked as answer]
If you can cut/paste plaintext examples into your issue it is easier for me to review them.

The "grouping" of the field data is due to `Export-Csv` attempting to convert those fields into a string value. This behavior isn't restricted to PSFalcon or the Spotlight APIs--it's how PowerShell works when you try to `Export-Csv`. 

Have you seen [Export-FalconReport](https://github.com/CrowdStrike/psfalcon/wiki/CSV-Output)? I created that command to make it easier to export to CSV. It does filter out some fields, but I tried to select what seemed like the relevant data from each request type. 

Try this and see if it works for you:
```
Get-FalconVulnerability -All -Detailed -Filter $Filter | Export-FalconReport -Path $vulnerabilities
```

**Comment by cjlane0289** [comment]
This worked perfectly.
Thank you!
